homeassistant/components/knx.py

```diff
@@ -249,3 +285,60 @@ def __init__(self, hass, device, hook, action, counter=1):
             hass.data[DATA_KNX].xknx, self.script.async_run,
             hook=hook, counter=counter)
         device.actions.append(self.action)
+
+
+class KNXExposeTime(object):
+    """Object to Expose Time/Date object to KNX bus."""
+
+    def __init__(self, xknx, expose_type, address):
+        """Initialize of Expose class."""
+        self.xknx = xknx
+        self.type = expose_type
+        self.address = address
+        self.device = None
+        self.register()
+
+    def register(self):
+        """Register listener."""
+        from xknx.devices import DateTime, DateTimeBroadcastType
+        broadcast_type_string = self.type.upper()
+        broadcast_type = DateTimeBroadcastType[broadcast_type_string]
+        self.device = DateTime(
+            self.xknx,
+            'Time',
+            broadcast_type=broadcast_type,
+            group_address=self.address)
+        self.xknx.devices.add(self.device)
+
+
+class KNXExposeSensor(object):
+    """Object to Expose HASS entity to KNX bus."""
+
+    def __init__(self, hass, xknx, expose_type, entity_id, address):
+        """Initialize of Expose class."""
+        self.hass = hass
+        self.xknx = xknx
+        self.type = expose_type
+        self.entity_id = entity_id
+        self.address = address
+        self.device = None
+        self.register()
+
+    def register(self):
+        """Register listener."""
+        from xknx.devices import ExposeSensor
+        self.device = ExposeSensor(
+            self.xknx,
+            name=self.entity_id,
+            group_address=self.address,
+            value_type=self.type)
+        self.xknx.devices.add(self.device)
+        async_track_state_change(
```

Comment:
This is an async method. That means that this method is only allowed to be called from an async context. This doesn't seem to be the case.